forui/lib/src/widgets/accordion.dart

```diff
@@ -0,0 +1,525 @@
+    await controller.reverse();
+    notifyListeners();
+  }
+}
+
+/// An [FAccordionController] that allows one section to be expanded at a time.
+class FRadioAccordionController extends FAccordionController {
+  /// Creates a [FRadioAccordionController].
+  FRadioAccordionController({
+    int? value,
+    super.duration,
+    super.min,
+    int super.max = 1,
+  }) : super(values: value == null ? {} : {value});
+
+  @override
+  Future<void> toggle(int index) async {
+    final toggle = <Future>[];
+    if (!(_values.length <= _min)) {
+      if (!_values.contains(index) && _values.length >= _max!) {
+        toggle.add(collapse(_values.first));
+        _values.remove(_values.first);
+      }
+    }
+    toggle.add(super.toggle(index));
+    await Future.wait(toggle);
+  }
+}
+
+/// A vertically stacked set of interactive headings that each reveal a section of content.
+class FAccordion extends StatefulWidget {
+  /// The controller.
+  ///
+  /// See:
+  /// * [FRadioAccordionController] for a single radio like selection.
+  /// * [FAccordionController] for default multiple selections.
+  final FAccordionController? controller;
+
+  /// The items.
+  final List<FAccordionItem> items;
+
+  /// The style. Defaults to [FThemeData.accordionStyle].
+  final FAccordionStyle? style;
+
+  /// Creates a [FAccordion].
+  const FAccordion({
+    required this.items,
+    this.controller,
+    this.style,
+    super.key,
+  });
+
+  @override
+  State<FAccordion> createState() => _FAccordionState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('controller', controller))
+      ..add(DiagnosticsProperty('style', style))
+      ..add(IterableProperty('items', items));
+  }
+}
+
+class _FAccordionState extends State<FAccordion> {
+  late final FAccordionController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = widget.controller ?? FAccordionController();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final style = widget.style ?? context.theme.accordionStyle;
+    return Column(
+      children: [
+        for (var i = 0; i < widget.items.length; i++)
+          _Item(
+            style: style,
+            index: i,
+            item: widget.items[i],
+            controller: _controller,
+          ),
+      ],
+    );
+  }
+}
+
+/// An item that represents a header in a [FAccordion].
+class FAccordionItem {
+  /// The title.
+  final Widget title;
+
+  /// The child.
+  final Widget child;
+
+  /// Whether the item is initially expanded.
+  final bool initiallyExpanded;
+
+  ///
+  FAccordionItem({required this.title, required this.child, this.initiallyExpanded = false});
+}
+
+/// An interactive heading that reveals a section of content.
+///
+/// See:
+/// * https://forui.dev/docs/accordion for working examples.
+class _Item extends StatefulWidget {
```

Comment:
Thinking about it, should we expose `_Item` instead, and pass in the index via an inherited widget? That way, we don't need the current FAccordionItem.